TransformationSystem/DB/TransformationDB.py

```diff
@@ -1525,3 +1521,94 @@ def __addDirectory( self, path, force ):
       else:
         successful.append( lfn )
     return {"OK":True, "Value": len( res['Value']['Successful'] ), "Successful":successful, "Failed": failed }
+
+  def setMetadata( self, path, usermetadatadict ):
+    """ It can be applied to a file or to a directory (path). For a file, add the file to Transformations if the updated metadata dictionary passes the filter.
+        For a directory, add the files contained in the directory to the Transformations if the the updated metadata dictionary passes the filter.
+    """
+    gLogger.info( "setMetadata: Attempting to set metadata %s to %s" % (usermetadatadict, path) )
+    successful = {}
+    failed = {}
+    if isinstance( path, dict ):
+      path = path.keys()[0]
+
+    transFiles = {}
+    filesToAdd = []
+
+    catalog = FileCatalog()
+    isFile = catalog.isFile( path )['Value']['Successful'][path]
+    isDirectory = catalog.isDirectory( path )['Value']['Successful'][path]
+
+    if isFile:
+      res = catalog.getFileUserMetadata( path )
+    elif isDirectory:
+      res = catalog.getDirectoryUserMetadata( path )
+
+    if not res['OK']:
+      failed[path] = res['Message']
+      return S_OK( {'Successful':successful, 'Failed':failed } )
```

Comment:
Yes, the method treats one element, as for the FileCatalogDB, which is either a file, either a directory. The reason why the method returns a dictionary of successful / failed is that, the effect of setMetadata in this case is to add to the transformations all the files that match the query conditions of the transformations. So the effect of setMetadata is on a set of files...
Does it make sense in this case to return a  successful / failed dictionary?